File: java/exception-handler-example-api/src/main/java/com/example/infrastructure/notifications/Notification.java
package com.example.infrastructure.notifications;

import com.example.infrastructure.notifications.Enums.NotificationType;

public class Notification {

    private NotificationType type;
    private String message;

    public Notification(NotificationType type, String message) {
        this.type = type;
        this.message = message;
    }

    public Notification(String message) {
        this.type = NotificationType.VALIDATION_ERROR;
        this.message = message;
    }

    public NotificationType getType() {
        return type;
    }

    public String getMessage() {
        return message;
    }

    public static Notification InexistentId(String message) {
        return new Notification(NotificationType.BUSINESS_ERROR, message);
    }

    public static Notification ValidationError(String message) {
        return new Notification(NotificationType.VALIDATION_ERROR, message);
    }
}

File: CSharp/ExceptionHandlerExample.Api/Infraestructure/Notifications/Notification.cs
using System.Net;
using Infra.CrossCutting.Notifications.Enums;

namespace Infra.CrossCutting.Notifications
{
    public class Notification
    {
        public NotificationType Type { get; private set; }
        public string Message { get; set; } 

        public Notification(NotificationType type, string message)
        {
            Type = type;
            Message = message;
        }

        public Notification(string message = null)
        {
            Type = NotificationType.VALIDATION_ERROR;
            Message = message;
        }

        public T InexistentId<T>(string message) where T : Notification
        {
            this.Type = NotificationType.BUSINESS_ERROR;
            this.Message = message;
            return (T)this;
        }

        public T ValidationError<T>(string message) where T : Notification
        {
            this.Type = NotificationType.VALIDATION_ERROR;
            this.Message = message;
            return (T)this;
        }
    }
}